napari/layers/labels/labels.py

```diff
@@ -329,6 +331,7 @@ def __init__(
         self._brush_size = 10
 
         self._selected_label = 1
+        self.prev_selected_label = None
```

Comment:
Should we keep this private?

Reply:
I initially thought about it, but as in `_labels_key_bindings.swap_background_color` we are trying to get access to this attribute outside of class, I decided to make it public.

Reply:
For that reason, I think `_background_label` should also be public.

Reply:
However, if we decide to implement it in a method of `Layer` as you suggested above, then we can make `prev_selected_label` private.

Reply:
Btw, @Czaki is it possible to make an attribute of a layer read-only without emitting any events. Initially, I wanted to make `prev_selected_label` public but make it read-only by implementing it as a property without implementing the setter. However, when you add a `property` to a layer, some code requires that it must have an `event` with the same name.

Reply:
I think that's good; if we expose it publicly, we should also make an event for it because we intend it to be used by users, and that's the standard for the rest of the codebase.

Reply:
If I understand correctly, it will break the linking of layers, as during linking this attribute will be attempted to be synchronized using the setter, which does not exist.

https://github.com/napari/napari/blob/95356e45ef7573e6e5f48abde0639388a0cea007/napari/layers/utils/_link_layers.py#L116-L133

Reply:
right... Then I stand by my advice of making this private altogether for now :P

though we should probably have some logic in there to check whether an attribute is settable.

Reply:
+1 to making it private.

Reply:
Yeah I think it seems like an implementation detail, so private 👍